### PR TITLE
Log inventory errors to Cloud Logging

### LINKED ISSUE #186: Build merge-to-main-cart on avocano-cart: FAILURE (Sold out!)
Cloud Build encountered an error: 

- Project: avocano-cart
- Trigger: merge-to-main-cart
- Commit: [52ebacf](https://github.com/glasnt/avocano/commit/52ebacf61685bcc67f83834fbdb07f5507d7e3b1) on branch main

[View Logs](https://console.cloud.google.com/cloud-build/builds/d376c9c3-fc50-43b2-a185-ff784510b064?project=751300358416&utm_campaign=google-cloud-build-notifiers&utm_medium=http&utm_source=google-cloud-build)

<sub>I'm an automation.  Googlers: go/avocano-admin-notifiers</sub>

### LINKED ISSUE #207: 💡 [Feature Request] Structured log for out of inventory warning
## Expected

A structured log is emitted for out of inventory errors that can be used as the basis for a log-based metric.

## Actual

An error is raised for API response: https://github.com/GoogleCloudPlatform/avocano/blob/88e8cc858e67f279a8a4501a3c76903a460ae4f6/server/store/serializers.py#L90-L98

There is no log entry.

### PR DESCRIPTION
**Pull request by glasnt**
## Description

Fixes #207

Fixes #186 

Alternative to #291

Errors in cases of:

Buy Now: product sells out
Buy Now: purchase attempted after a sell out
Cart: purchase exceeds available inventory.

Uses json with `severity` and `message` keys, plus a new extras, to print to stdout, which is then picked up by Cloud Logging when run in a Cloud Run deployment. Still prints to stdout in localhost/ci environments

Screenshot 2023-07-14 at 2 22 39 pm

### CODE REVIEW

File: server/store/views.py [hunk 1]
```diff
@@ -47,6 +47,23 @@ class ProductPurchaseException(APIException):
     }
 
 
+def log_error(error_name, error_message, product):
```
Comment (grayside) [inline server/store/views.py:50]: discussion: This logging helper seems a bit specialized, but I think that keeps it approachable to read and infer more abstraction.

File: server/store/views.py [hunk 2]
```diff
@@ -47,6 +47,23 @@ class ProductPurchaseException(APIException):
...
+    # Log error by writing structured JSON. Can be then used with log-based alerting, metrics, etc.
+    print(
+        json.dumps(
```
Comment (grayside) [inline server/store/views.py:53]: discussion: We're not adding the trace to this log, which means it won't have cross-reference to Cloud Tracing. I think that's fair. I think my preference would be to gain that by upgrading to a cloud logging library that has low overhead but handles that for us.

**Review by grayside — APPROVED** [review]

File: server/store/views.py (hunk 2 above)
Comment (glasnt) [reply server/store/views.py:53]: I may be able manually add tracing per https://cloud.google.com/run/docs/logging#writing_structured_logs, but I would need to confirm I have access to the request in the view/serialiser contexts. I presumed the original ask in 207 didn't include tracing, but I can adapt this PR to suit.

Comment (glasnt) [reply server/store/views.py:53]: As discussed, this is fine for now. Adding in tracing is a future feature

**Pull request merged**